src/faq.md

```diff
@@ -79,7 +79,62 @@ outputs:
       outputEval: ${self[0].basename=inputs.otu_table_name; return self;}
 ```
 
-## Setting `self`-based input bindings for optional inputs
+## Referencing a Local Script
+
+There are two ways to reference a local script:
+
+The first method involves adding the folder containing your scripts to the `PATH` environment variable. This allows you to run the shell script without using `sh` or `bash` commands.
+
+Start with adding a _shebang_ at the top of your file:
+
+```{code-block}
+#!/bin/bash
+```
+
+After that, make the script executable with the command `chmod +x scriptname.sh`
+
+Finally, modify your `PATH` to add the directory where your script is located.
+(It is good practice to use `$HOME/bin` for storing your own scripts).
+
+```bash
+export PATH=$PATH:/appropriate/directory
+```
+
+Now you can use `baseCommand: scriptname.sh` to run the script directly.
+
+```cwl
+#!/bin/bash
+cwlVersion: v1.0
+class: CommandLineTool
+baseCommand: scriptname.sh
+```
+
+When you wish to share your work later, you can place your script in a software container in the Docker format.
+
+The second method involves including an input of `type: File` in the script itself:
+
+```cwl
+class: CommandLineTool
+
+inputs:
+  my_script:
+     type: File
+     inputBinding:
+        position: 0
+
+
+  # other inputs go here
+
+baseCommand: sh
+
+outputs: []
+```
+
+```{note}
+In CWL, everything must be directly stated.
```

Comment:
I don't understand what you want to add here.

Reply:
I believe this was taken from rephrasing this comment of yours -- 

"In CWL we have to be explicit about everything, so either the script needs to be

    available on the system PATH (for cwltool doing local execution, no --preserve-environment needed)

or

    part of the Docker format software container (either a fixed path or again on the system PATH defined in the container)

or

    it needs to be an input to the CommandLineTool description itself (as shown in my previous comment)

InitialWorkDirRequirement could be used to arrange the input data alongside one or more scripts (themselves also inputs) if they expect to be in the same folder; is that what you're asking for?"

Reply:
Hmm, this comment seems a bit lost as it is right now. IMHO it could be re-written to better explain what it means, or removed. Will add a note to the staging PR.